app/controllers/course/assessment/submissions_controller.rb

```diff
@@ -108,17 +56,18 @@ def authorize_submission!
     end
   end
 
-  def load_or_create_answers
-    return unless @submission.attempting?
+  def load_service
```

Comment:
But in the second case, why not just `delegate :update to: :@service` and make `@service` a service instance ? As what I implemented now.  

I can put `extract_instance_variables` in the end of the around action, and then the service don't need to know the controller.